secret_store/src/key_server_cluster/math.rs

```diff
@@ -434,7 +434,27 @@ pub mod tests {
 		joint_public: Public,
 	}
 
-	fn run_key_generation(t: usize, n: usize, id_numbers: Option<Vec<Secret>>) -> KeyGenerationArtifacts {
+	struct ZeroGenerationArtifacts {
+		polynoms1: Vec<Vec<Secret>>,
+		secret_shares: Vec<Secret>,
+	}
+
+	fn prepare_polynoms1(t: usize, n: usize, secret_required: Option<Secret>) -> Vec<Vec<Secret>> {
+		let mut polynoms1: Vec<_> = (0..n).map(|_| generate_random_polynom(t).unwrap()).collect();
+		// if we need specific secret to be shared, update polynoms so that sum of their free terms = required secret
+		if let Some(mut secret_required) = secret_required {
+			for polynom1 in polynoms1.iter_mut().take(n - 1) {
+				let secret_coeff1 = generate_random_scalar().unwrap();
+				secret_required.sub(&secret_coeff1).unwrap();
```

Comment:
is that tested to wrap around?

Reply:
Thanks for reviewing! Yes - it will wrap around in case of underflow. Please see `run_zero_key_generation` - 0 is passed here as `secret_required` => any non-zero value (which is generated by `generate_random_scalar`) will cause underflow => yes it is tested.